card/mod/03_machines/lib/javascript/wagn.js.coffee

```diff
@@ -23,19 +23,23 @@ $.extend wagn,
         wagn.slotParams value, processed, cgiKey
       else
         processed[cgiKey] = value
-        
+
   slotReady: (func)->
     $('document').ready ->
       $('body').on 'slotReady', '.card-slot', (e) ->
         e.stopPropagation()
         func.call this, $(this)
   pingName: (name, success)->
-    $.getJSON wagn.rootPath + '/', { format: 'json', view: 'status', 'card[name]': name }, success  
+    $.getJSON wagn.rootPath + '/', { format: 'json', view: 'status', 'card[name]': name }, success
 
 jQuery.fn.extend {
-  slot: -> 
-    if @attr('slotSelector')
-      @closest @attr('slotSelector')
+  slot: ->
+    if @data('slot-selector')
+      close = @closest(@data('slot-selector'))
+      if close.length == 0
+        $('body').find(@data('slot-selector'))
+      else
```

Comment:
Is it safe to assume that the the slot-selector is always a sibling?  If so, you could go back to the parent slot rather than to body, as Gerry suggests.  It does seem like it would be nice if we had some other relative alternatives.